jetson/db/db_connection.py

```diff
@@ -0,0 +1,125 @@
+    Class name must match exactly the spelling of the corresponding table in database
+
+    To insert into database table using this class, use this class as input for function sql_insert:
+        sql_insert(IMAGE('image_name', 'image_date', 'image_time', 'init_vector'))
+
+        Args:
+            image_name (string) : name of image
+            image_date (datetime obj) : date image was taken
+            image_time (datetime obj) : time image was taken
+            init_vector (string) : decryption key for encrypted image
+    """
+
+    def __init__(self, image_name, image_date, image_time, init_vector):
+        self.Image_Name = image_name
+        self.Image_Date = image_date
+        self.Image_Time = image_time
+        self.Init_Vector = init_vector
+
+
+class BBOX(Table):
+    """Class defines the BBox table in the database with init inputs being the columns in the table.
+    Class name must match exactly the spelling of the corresponding table in database
+
+    To insert into database table using this class, use this class as input for function sql_insert:
+        sql_insert(BBOX(xmin, ymin, xmax, ymax, conf, goggles))
+
+        Args:
+            xmin (float) : lower left x-coordinate of bounding box
+            ymin (float) : lower left y-coordinate of bounding box
+            xmax (float) : upper right x-coordinate of bouding box
+            ymax (float) : upper right y-coordinate of bounding box
+            conf (float) : confidence score for bounding box
+            goggles (bool) : whether goggles were detected in bounding box
+            image_name (string): name of image that contains the bounding box
+    """
+
+    def __init__(self, xmin, ymin, xmax, ymax, conf, goggles, image_name):
+        self.X_Min = xmin
+        self.Y_Min = ymin
+        self.X_Max = xmax
+        self.Y_Max = ymax
+        self.Confidence = conf
+        self.Goggles = goggles
+        self.Image_Name = image_name
+
+
+@contextmanager
+def sql_connection():
+    """Sets up connection to mysql database
+
+    Yields:
+        sql connection: sql connector object to the sql database
+    """
+    connection = mysql.connector.connect(
+        host=config.SQL_HOST,
+        user=config.USER_NAME,
+        password=config.PASSWORD,
+        database=config.KEYSPACE
+    )
+
+    with closing(connection) as connection:
+        yield connection
+
+
+@contextmanager
+def sql_cursor():
+    """Gets sql cursor from database connection
+
+    Yields:
+       sql cursor : cursor object to database
+    """
+    with sql_connection() as connection:
+        yield connection.cursor(buffered=True)
+        connection.commit()
+
+
+def sql_insert(table: Table):
+    """Inserts row of information for a specified table in database
+
+    Args:
+        table (class obj): Class with name that corresponds to the table for data to be inserted into,
+                            class name must match the database table name exactly

```

Comment:
"class name must match the database table name exactly" is redundant since it is defined in the class definition docstring